src/components/Modal.js

```diff
@@ -60,16 +61,36 @@ const Modal = (props) => {
             <SafeAreaInsetsContext.Consumer>
                 {(insets) => {
                     const {paddingTop, paddingBottom} = getSafeAreaPadding(insets);
+
+                    /**
+                     * Calculates the real top and bottom padding of the container,
+                     * given the given padding, the plataform specific safe area padding,
+                     * and if it should sum both values or not
```

Comment:
I'm unsure of what "real" means in this case? It looks like we are either adding the safe area padding to the existing padding of a container or not. I think this actually might be a bit clearer if we just handled this inline as it took some effort for me to figure out why we needed this method. 

Perhaps something like this is OK ?

```js
{
    paddingTop: shouldAddSafeAreaPaddingTop 
        ? modalContainerStyle.paddingTop + safeAreaPaddingTop 
        : modalContainerStyle.paddingTop,
}
```

Reply:
My poor choice of words always hunts me :sweat_smile:.

You are right, inline looks good. The only problem with that is currently `modalContainerStyle.paddingTop` and `modalContainerStyle.paddingBottom` could not be defined (not be explicitly expressed in `getModalStyle.js`). The function was taking care of this case.

The option around that I like is checking this in `getModalStyle.js`, before the `return` statement, and maintaining the code you wrote:

**getModalStyle.js**
```javascript
    if (!modalContainerStyle.paddingTop) {
        modalContainerStyle = {...modalContainerStyle, paddingTop: 0};
    }
    if (!modalContainerStyle.paddingBottom) {
        modalContainerStyle = {...modalContainerStyle, paddingBottom: 0};
    }
```

The other option would be

**Modal.js** 
```javascript
paddingTop: shouldAddTopSafeAreaPadding
    ? (modalContainerStyle.paddingTop || 0) + safeAreaPaddingTop
    : modalContainerStyle.paddingTop || 0,
paddingBottom: shouldAddBottomSafeAreaPadding
    ? (modalContainerStyle.paddingBottom || 0) + safeAreaPaddingBottom
    : modalContainerStyle.paddingBottom || 0,
```